server/src/main/java/org/opensearch/gateway/PrimaryShardAllocator.java

```diff
@@ -386,11 +422,27 @@ protected static NodeShardsResult buildNodeShardsResult(
             }
         }
 
-        /*
-          Orders the active shards copies based on below comparators
-          1. No store exception i.e. shard copy is readable
-          2. Prefer previous primary shard
-          3. Prefer shard copy with the highest replication checkpoint. It is NO-OP for doc rep enabled indices.
+        nodeShardStates.sort(createActiveShardComparator(matchAnyShard, inSyncAllocationIds));
+
+        if (logger.isTraceEnabled()) {
+            logger.trace(
+                "{} candidates for allocation: {}",
+                shard,
+                nodeShardStates.stream().map(s -> s.getNode().getName()).collect(Collectors.joining(", "))
+            );
+        }
+        return new NodeShardsResult(nodeShardStates, numberOfAllocationsFound);
+    }
+
+    private static Comparator<NodeGatewayStartedShards> createActiveShardComparator(
```

Comment:
this is a private method anyway, why did you need to refactor it?

Reply:
This was not necessary, made this change to make the code more readable and easy to understand.